app/gui/view/graph-editor/src/component/node/input/widget.rs

```diff
@@ -158,12 +166,76 @@ impl SingleChoice {
             });
 
             let focus_in = display_object.on_event::<event::FocusIn>();
-            let focus_out = display_object.on_event::<event::FocusOut>();
-            is_focused <- bool(&focus_out, &focus_in);
-
-            dropdown.set_open <+ is_focused;
+            eval focus_in((_) dropdown.borrow_mut().initialize_on_open());
         }
 
         Self { display_object, frp, dropdown, activation_dot }
     }
 }
+
+
+
+// ====================
+// === LazyDropdown ===
+// ====================
+
+/// A lazy dropdown that is only initialized when it is opened for the first time. This prevents
+/// very long initialization time, as dropdown view creation is currently a very slow process.
+///
+/// Once grid-view creation is reasonably fast, this might be replaced by direct dropdown
+/// initialization on widget creation.
```

Comment:
Comments should explain semantically what code does. This is not a semantic explanation - rather a bug description. It can be left here, but it should be written in a form of a FIXME with a connected issue. Otherwise, it's impossible to track such comments in the code and remove them after the given issue is fixed. Also, sentences such as "is currently a very slow process" are very easy to get obsolete and no one will remember to update the comment.

Reply:
Added a fixme with a connected issue.